odoo/addons/base/transpiler/js_transpiler.py

```diff
@@ -0,0 +1,467 @@
+
+    .. code-block:: javascript
+
+        // before
+        export default something =
+        // after
+        __exports[Symbol.for("default")] =
+    """
+
+    new_content = convert_export_function_or_class(content, True)
+    new_content = convert_variable_export(new_content, True)
+    p = re.compile(r'^(?P<space>\s*)export\s+default(\s+\w+\s*=)?', re.MULTILINE)
+    repl = r"""\g<space>__exports[Symbol.for("default")] ="""
+    return p.sub(repl, new_content)
+
+
+def convert_basic_import(content):
+    """
+    Transpile the simpler import call.
+
+    .. code-block:: javascript
+
+        // before
+        import { a, b, c as x } from "some/path"
+        // after
+        const {a, b, c: x} = require("some/path")
+    """
+    def repl(matchobj):
+        new_list = matchobj["list"].replace(" as ", ": ")
+        return f"{matchobj['space']}const {new_list} = require({matchobj['path']})"
+
+    p = re.compile(r"^(?P<space>\s*)import\s+(?P<list>{(\s*\w+\s*,?\s*)+})\s*from\s*(?P<path>(\".*\")|('.*')|(`.*`))", re.MULTILINE)
+    return p.sub(repl, content)
+
+
+def convert_legacy_default_import(content):
+    """
+    Transpile legacy imports (that were used as they were default import).
+    Legacy imports means that their name is not a path but a <addon_name>.<module_name>.
+    It requires slightly different processing.
+
+    .. code-block:: javascript
+
+        // before
+        import module_name from "addon.module_name"
+        // after
+        const module_name = require("addon.module_name")
+    """
+    p = re.compile(r"^(?P<space>\s*)import\s+(?P<identifier>\w+)\s*from\s*(?P<path>(\"\w+\.\w+\")|('\w+\.\w+')|(`\w+\.\w+`))", re.MULTILINE)
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)"""
+    return p.sub(repl, content)
+
+
+def convert_default_import(content):
+    """
+    Transpile the default import call.
+
+    .. code-block:: javascript
+
+        // before
+        import something from "some/path"
+        // after
+        const something = require("some/path")[Symbol.for("default")]
+    """
+    p = re.compile(r"^(?P<space>\s*)import\s+(?P<identifier>\w+)\s*from\s*(?P<path>(\".*\")|('.*')|(`.*`))", re.MULTILINE)
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)[Symbol.for("default")]"""
+    return p.sub(repl, content)
+
+
+def convert_relative_require(url, content):
+    """
+    Convert the relative require call that was generated by convert_unnamed_relative_import.
+    Simply give it the new path system.
+    Note: if the relative path uses the legacy system (addon.module) we keep it as it is valid.
+
+    .. code-block:: javascript
+
+        // before
+        require "some/path"
+        // after
+        require "@module/path"
+    """
+    new_content = content
+    p = re.findall(r"""require\((["'])([^@\"\']+)(["'])\)""", new_content)
+    for open_quote, path, close_quote in p:
+        if not bool(re.match(r"\w+\.\w+", path)):
```

Comment:
Python idiom: don't use `bool` with `if` and logic operators, they already do it
```suggestion
        if not re.match(r"\w+\.\w+", path):
```